xbmc/pvr/addons/PVRClients.cpp

```diff
@@ -935,76 +966,80 @@ bool CPVRClients::RenameChannel(const CPVRChannelPtr &channel)
   return (error == PVR_ERROR_NO_ERROR || error == PVR_ERROR_NOT_IMPLEMENTED);
 }
 
-bool CPVRClients::IsKnownClient(const AddonPtr &client) const
+bool CPVRClients::IsKnownClient(const BinaryAddonBasePtr& client) const
 {
   // database IDs start at 1
   return GetClientId(client) > 0;
 }
 
 void CPVRClients::UpdateAddons(void)
 {
-  VECADDONS addons;
-  CAddonMgr::GetInstance().GetInstalledAddons(addons, ADDON_PVRDLL);
+  BinaryAddonBaseList addonInfos;
+  CServiceBroker::GetBinaryAddonManager().GetAddonInfos(addonInfos, false, ADDON_PVRDLL);
 
-  if (addons.empty())
+  if (addonInfos.empty())
     return;
 
-  for (auto &addon : addons)
+  for (auto &addon : addonInfos)
   {
-    bool bEnabled = !CAddonMgr::GetInstance().IsAddonDisabled(addon->ID());
+    bool enabled = CServiceBroker::GetBinaryAddonManager().IsAddonEnabled(addon->ID(), ADDON_PVRDLL);
+    UpdateAddon(addon, enabled);
+  }
 
-    if (bEnabled && (!IsKnownClient(addon) || !IsCreatedClient(addon)))
-    {
-      std::hash<std::string> hasher;
-      int iClientId = static_cast<int>(hasher(addon->ID()));
-      if (iClientId < 0)
-        iClientId = -iClientId;
+  CServiceBroker::GetPVRManager().Start();
```

Comment:
Yeah,  I have seen this, it however would be nice if we could get rid off this circular dependency.